core-meta.html

```diff
@@ -68,12 +68,54 @@
          * @type String
          * @default 'default'
          */
-        type: String
+        type: String,
+
+        /**
+         * The value of the meta-data to store. If no value is specified, the
+         * value defualts to a self reference (`this`).
+         *
+         * @attribute value
+         * @type *
+         * @default this
+         */
+        value: String,
+
+        /**
+         * The key used to store the `value` under the `type` namespace.
+         * Defaults to the `id` value.
+         *
+         * @attribute key
+         * @type String
+         * @default this.id
+         */
+        key: String
```

Comment:
I think that `key` is a more natural choice for a custom element because it can be potentially bound to another value. This cannot be done with `id` (even though it is possible to detect when it is changed).